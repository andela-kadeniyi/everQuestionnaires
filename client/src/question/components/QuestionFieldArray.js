import React from 'react';
import { Field } from 'redux-form';
import renderField from '../../commonComponents/Field';

const QuestionFieldArray = ({ fields }) => (
  <ul>
    <li>
      <button type="button" onClick={() => fields.push({})}>Add Question</button>
    </li>
    {fields.map((question, index) => (
      <li key={index}>
        <button
          type="button"
          title="Remove Question"
          onClick={() => fields.remove(index)}
        />
        <h4>Question #{index + 1}</h4>
        <Field
          name={`${question}.name`}
          type="text"
          component={renderField}
          label="Name"
        />
        <Field
          name={`${question}.label`}
          type="text"
          component={renderField}
          label="Label"
        />
      </li>
    ))}
  </ul>
);


export default QuestionFieldArray;