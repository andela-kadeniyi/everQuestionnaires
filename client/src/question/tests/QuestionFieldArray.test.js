import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import QuestionFieldArray from '../../question/components/QuestionFieldArray';

describe('QuestionnaireForm', () => {
  it('renders self without crashing', () => { 
    const component = shallow(<QuestionFieldArray fields={[]}  />)
    expect(toJson(component)).toMatchSnapshot();
  });
})