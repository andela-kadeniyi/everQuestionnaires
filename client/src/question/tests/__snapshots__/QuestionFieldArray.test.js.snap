// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireForm renders self without crashing 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
      type="button"
    >
      Add Question
    </button>
  </li>
</ul>
`;
